apps/keyboard/js/keyboard/input_method_manager.js

```diff
@@ -448,25 +408,16 @@ InputMethodManager.prototype.activateIMEngine = function(imEngineName) {
   this.app.console.log(
     'InputMethodManager.activateIMEngine()', imEngineName);
 
-  // dataPromise is the one we previously created with updateInputContextData()
-  var dataPromise = this._inputContextData;
-
-  if (!dataPromise) {
-    console.warn('InputMethodManager: activateIMEngine() called ' +
-      'without calling updateInputContextData() first.');
-  }
-
   // Create our own promise by resolving promise from loader and the passed
   // dataPromise, then do our things.
   var loaderPromise = this.loader.getInputMethodAsync(imEngineName);
   var settingsPromise = this.imEngineSettings.initSettings();
 
-  var p = Promise.all([loaderPromise, dataPromise, settingsPromise])
+  var p = Promise.all([loaderPromise, settingsPromise])
```

Comment:
Remove inputContext promise, since we can get text from this.app.inputContext.text in synchronous way.